### ISSUE TITLE
[FEATURE REQUEST] Allow multiple rules to apply to a domain

### ISSUE THREAD

**Issue by d07434b2c08f** [mrdokenny/Cookie-AutoDelete#912]
**Is your feature request related to a problem? Please describe.** 
I'm trying to set up my rules so there's a \*.\* greylist to clear Cache and IndexedDB for all domains on browser exit, but also have this apply to domains for which I'm saving cookies. For example, if I have \*.google.com set up as a whitelist to save cookies but not save cache, my assumption was that when I closed all my Google tabs, the cookies will be preserved indefinitely, and the cache would be cleared but not until I closed my browser. Instead, the cache gets cleared when I close the Google tabs because it's not included in the Google whitelist.

**Describe the solution you'd like** 
I'd like Cookie AutoDelete to iterate through all the rules that apply to the current domain and apply them (or not) based on their specificity. I can see both \*.google.com and \*.\* in the menu popup, so I'm not sure if it's already supposed to do that or if this is all just a bug.

**Describe alternatives you've considered** 
None as far as I can tell, besides deciding between keeping cache indefinitely or deleting it sooner than I'd like.

**Additional context** 
![example](https://user-images.githubusercontent.com/55886206/94558286-fe141180-021c-11eb-97af-c0a243fb6aa5.png)

**Comment by kennethtran93** [comment]
Duplicate of #735
